src/Arcus.Testing.Tests.Integration/Integration/DataFactory/Fixture/TemporaryDataFactoryDataFlow.cs

```diff
@@ -370,4 +409,94 @@ public async ValueTask DisposeAsync()
             }
         }
     }
+
+    public class TempDataFlowOptions
+    {
+        public TempDataFlowSourceOptions Source { get; } = new();
+        public TempDataFlowSinkOptions Sink { get; } = new();
+    }
+
+    public class TempDataFlowSourceOptions
+    {
+        /// <summary>
+        /// Gets the unique key and values of the parameters of the source DataSet of the temporary DataFlow in Azure DataFactory.
+        /// </summary>
+        public IDictionary<string, string> SourceDataSetParameterKeyValues { get; } = new Dictionary<string, string>();
+
+        public TempDataFlowSourceOptions AddFolderPathParameters(IDictionary<string, string> parameters)
+        {
+            foreach (var parameter in parameters)
+            {
+                SourceDataSetParameterKeyValues.Add(parameter.Key, parameter.Value);
+            }
+            return this;
+        }
+
+        internal void ApplyOptions(ref DelimitedTextDataset dataSet, string sourceDataSetName)
+        {
+            string folderPathExpression = $"@concat('{sourceDataSetName}/', ";
+            foreach (var sourceDataSetParametersKey in SourceDataSetParameterKeyValues.Keys)
+            {
+                folderPathExpression += $"dataset().{sourceDataSetParametersKey}, '/', ";
+            }
+            // Remove the last ", '/', " part only if we have parameters
+            // Else just remove the last ", " part
+            folderPathExpression = (SourceDataSetParameterKeyValues.Keys.Count > 0 ? folderPathExpression[..^7] : folderPathExpression[..^2]) + ")";
+
+            if (SourceDataSetParameterKeyValues.Any())
+            {
+                dataSet.DataLocation.FolderPath = DataFactoryElement<string>.FromExpression(folderPathExpression);
+            }
+        }
+
+        internal void AddDataSetParameters(ref DelimitedTextDataset dataSet, IDictionary<string, string> sourceDataSetParameterKeyValues)
```

Comment:
```suggestion
        internal void ApplyOptions(DelimitedTextDataset dataSet, string sourceDataSetName)
        {
            string folderPathExpression = $"@concat('{sourceDataSetName}/', ";
            foreach (var sourceDataSetParametersKey in SourceDataSetParameterKeyValues.Keys)
            {
                folderPathExpression += $"dataset().{sourceDataSetParametersKey}, '/', ";
            }
            // Remove the last ", '/', " part only if we have parameters
            // Else just remove the last ", " part
            folderPathExpression = (SourceDataSetParameterKeyValues.Keys.Count > 0 ? folderPathExpression[..^7] : folderPathExpression[..^2]) + ")";

            if (SourceDataSetParameterKeyValues.Any())
            {
                dataSet.DataLocation.FolderPath = DataFactoryElement<string>.FromExpression(folderPathExpression);
            }
        }

        internal void AddDataSetParameters(DelimitedTextDataset dataSet, IDictionary<string, string> sourceDataSetParameterKeyValues)
```
Objects are already passed by reference.